lib/service_manager.dart

```diff
@@ -383,6 +386,30 @@ class ServiceExtensionManager {
     }
   }
 
+  Future<void> _restoreExtensionFromDevice(String name) async {
+    if (!extensions.statefulExtensionsWhitelist.contains(name)) {
```

Comment:
if statefulExtensionsWhitelist was instead a map from 
String extension name to ServiceExtensionDescription you would know what the expected type of the response is by the types of the values for trueValue and falseValue in the extension description.

Reply:
done.